### PR TITLE
PLAT-103496: Support localization for Agate components

### PR DESCRIPTION
**Pull request by Baekwoo-s**
Support localization for Agate components

Enact-DCO-1.0-Signed-off-by: Baekwoo Jung (baekwoo.jung@lge.com)

### CODE REVIEW

File: Spinner/Spinner.js [hunk 1]
```diff
@@ -21,6 +21,7 @@ import UiSpinnerBase from '@enact/ui/Spinner';
 import Skinnable from '../Skinnable';
 
 import componentCss from './Spinner.module.less';
+import $L from '../../moonstone/internal/$L/$L';
```
[the remark this answers is not in the document]
Comment (baekwoo) [reply Spinner/Spinner.js]: moonstone is wrong. I modified it to use agate/internal/$L.

[the remark this answers is not in the document]
Comment (Baekwoo-s) [reply Spinner/Spinner.js]: moonstone is wrong. I modified it to use agate/internal/$L.

**Comment by seunghoh** [comment]
Let's change DatePicker with iLib formatter first

**Comment by seunghoh** [comment]
LGTM.
For recording our discussion, we will implement DatePicker with iLib format in the other ticket

File: Spinner/Spinner.js [hunk 2]
```diff
@@ -55,7 +56,7 @@ const SpinnerCore = kind({
 	},
 
 	defaultProps: {
-		type: 'searching'
+		type: $L('searching')
```
Comment (0x64) [inline Spinner/Spinner.js:59]: This looks weird. This is a _public_ prop and the value should be a plane text. This value is used just to test like `type === 'searching'.

File: DateTimePicker/DateTimePicker.js [hunk 3]
```diff
@@ -67,7 +59,7 @@ class DateTimePickerBase extends React.Component {
 		this.state = {
 			hour: '12',
 			minute: '00',
-			meridiem: 'AM',
+			meridiem: $L('AM'),
```
Comment (0x64) [inline DateTimePicker/DateTimePicker.js:62]: This change also affects the value to be delivered to the app through `onChange`, `onSave`, and `onClose` callback props. Is this intended?

Comment (Baekwoo-s) [reply DateTimePicker/DateTimePicker.js:62]: I agree with your opinion. Fixed to compare type with plane text.

File: Spinner/Spinner.js (hunk 2 above)
Comment (Baekwoo-s) [reply Spinner/Spinner.js]: I agree with your opinion. Fixed to compare type with plane text.

File: DateTimePicker/DateTimePicker.js (hunk 3 above)
Comment (Baekwoo-s) [reply DateTimePicker/DateTimePicker.js:62]: In this code, there is a problem that ilib date is not passed to App developer.
Any changes to this code will be revised in PLAT-104574.

**Pull request merged**